samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -84,8 +86,50 @@ public static ZkConnection createZkConnection(String zkConnectString, int sessio
     return new ZkConnection(zkConnectString, sessionTimeoutMs);
   }
 
-  public static ZkClient createZkClient(ZkConnection zkConnection, int connectionTimeoutMs) {
-    return new ZkClient(zkConnection, connectionTimeoutMs);
+  /**
+   * helper method to create zkClient
+   * @param connectString - zkConnect string
+   * @param sessionTimeoutMS - session timeout
+   * @param connectionTimeoutMs - connection timeout
+   * @return zkClient object
+   */
+  public static ZkClient createZkClient(String connectString, int sessionTimeoutMS, int connectionTimeoutMs) {
```

Comment:
I moved both createClient and createPath to the CoordinationUtilsFactory, where it fits better.